EasyVCR/Censors.cs

```diff
@@ -189,45 +220,136 @@ internal string ApplyBodyParametersCensors(string body, ContentType? contentType
         internal IDictionary<string, string> ApplyHeaderCensors(IDictionary<string, string> headers)
         {
             if (headers.Count == 0)
+            {
                 // short circuit if there are no headers to censor
                 return headers;
+            }
 
             return _headersToCensor.Count == 0 ? headers : headers.ToDictionary(header => header.Key, header => ElementShouldBeCensored(header.Key, _headersToCensor) ? _censorText : header.Value);
         }
 
         /// <summary>
-        ///     Censor the appropriate query parameters.
+        ///     Censor the appropriate path elements and query parameters.
         /// </summary>
         /// <param name="url">Full URL string to apply censors to.</param>
         /// <returns>Censored URL string.</returns>
-        internal string? ApplyQueryParametersCensors(string? url)
+        internal string? ApplyUrlCensors(string? url)
         {
-            if (_queryParamsToCensor.Count == 0)
+            if (url == null)
+            {
+                // short circuit if url is null
+                return url;
+            }
+
+            if (_queryParamsToCensor.Count == 0 && _pathElementsToCensor.Count == 0)
             {
                 // short circuit if there are no censors to apply
                 return url;
             }
 
+            var uri = new Uri(url);
+
+            var path = uri.GetLeftPart(UriPartial.Path);
```

Comment:
Did we name `GetLeftPart`? If so, please rename this to what I assume should be something like `GetScheme` heh, "LeftPart" isn't really descriptive.

Or if that's not even it, you can now see my confusion I hope. This should be explicitly named for what exactly we are getting.

Reply:
Unfortunately, that's not our function, something built into URI in .NET. I'll add a comment.

Reply:
Gosh dang .NET... at least it's not us 😉